src/mgmt/srv_drpc.c

```diff
@@ -933,13 +937,23 @@ ds_mgmt_drpc_pool_reintegrate(Drpc__Call *drpc_req, Drpc__Response *drpc_resp)
 		return;
 	}
 
+	if (req->n_tierbytes == 0 || req->n_tierbytes > DAOS_MEDIA_MAX) {
+		D_ERROR("Invalid number of storage tiers: "DF_U64"\n",
+			req->n_tierbytes);
+		D_GOTO(out, rc = -DER_INVAL);
+	}
```

Comment:
so we get these sizes from yaml or parameters?  Probably it can try to query these information from PS leader if it fails. what do you think? or I misunderstood sth here.

Reply:
Here we get the sizes from the MS DB, who stores the original parameters from "dmg pool create"; we do not require "dmg pool reintegrate" to specify the sizes. The PS does not store these sizes, AFAIK.

Reply:
@wangdi1, does my previous comment answer your questions?

Reply:
ah, yes, It is  from MS DB, I see. thanks for clarification!

Reply:
Thanks. I'll ask for landing then.